modopt/signal/wavelet.py

```diff
@@ -153,7 +153,7 @@ def call_mr_transform(data, opt='', path='./',
                            ''.format(executable, stdout.rstrip('\n')))
 
     # Retrieve wavelet transformed data.
-    result = fits.getdata(file_mr)
+    result = fits.getdata(file_mr).astype(float)
```

Comment:
I don't understand why this constraint and also why now? Can we not get a complex image and thus complex wavelet coefficients?
Does astropy handle this better than scipy?

Reply:
@chaithyagr I see your point. I can replace `float` with `data.dtype` to be more flexible. Anyway, the previous issue was because the `dtype` returned by `fits.getdata` was `>f4` and the latest version of Scipy did not like that.